### ISSUE TITLE
DOCUMENT as injection token dependency example

### ISSUE THREAD

**Issue by goloti** [testing-library/angular-testing-library#186]
Would be nice if you can provide the same 10-inject-token-dependency example but injecting the DOCUMENT token instead, This token is required if you try to access browser's document global variable when you are using SSR with Angular Universal.

**Comment by timdeschryver** [comment]
Hi @goloti , what would you be looking for exactly?
It would be the same, but with the `DOCUMENT` token.

**Comment by goloti** [comment]
Yes, I'm sorry. I didn't provide any information or context... Let me try again.

Im using the DOCUMENT token to obtain the current URL with angular universal.

```typescript
constructor(@Inject(DOCUMENT) private _doc: Document) {
    this.currentURL = this._doc.location.pathname;
}
```

I have injected the DOCUMENT token in the same way that you are providing the DATA token. 
In the component, I'm just using `location.pathname`, So I thought that mocking just these properties would be enough. 


```typescript
  it('should create the app', async () => {
    await render(AppComponent, {
      providers: [
        {
          provide: DOCUMENT,
          useValue: {
            location: {
              pathname: '/'
            }
          }
        },
      ],
    });
    expect(screen.getByText(/hello/i)).toBeTruthy();
  });
```

However it throws this error:

`this._doc.querySelectorAll is not a function` 

I added a mock for that too. But it thows another error. I have continued providing mocks as long as I could (2-3 more times). But now I'm stuck with this error:

`The selector "#root0" did not match any elements`

I have prepared a little example for you. It contains the minimal code for reproducing the issue and my progress with the DOCUMENT mock:

https://github.com/goloti/atl-sample

You can find the test in [src/app/app.component.spec.ts](https://github.com/goloti/atl-sample/blob/master/src/app/app.component.spec.ts) file.

Could you help me with this?

**Comment by timdeschryver** [comment]
Hi, thanks a lot for the details and the reproduction!
I'm not familiar with SSR, but have you tried providing the "real" document as value?
Something like this:

```
await render(AppComponent, {
      providers: [
        {
          provide: DOCUMENT,
          useValue: document
        },
      ],
    });
```

**Comment by goloti** [comment]
Yes! It works! I have focused in mock implementation forgetting completely the real one...
Thank you very much!
